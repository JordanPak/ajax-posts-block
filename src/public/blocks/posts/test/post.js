/**
 * Test <Post> component
 *
 * @since 1.0.0
 */
import { shallow } from 'enzyme';

import Post from '../post';

global.apbHelper = { i18n: { by: 'by' } };

describe( 'Post Compoonent', () => {
	it( 'Shows post with all features correctly', () => {
		// mock a thumbnail
		const featuredMedia = [
			{
				alt_text: 'Thumbnail alt text',
				media_details: {
					sizes: {
						medium: {
							source_url: 'thumbnail.jpg',
							width: '400',
							height: '200',
						},
					},
				},
			},
		];

		const post = shallow(
			<Post
				title={ { rendered: 'Jimbo Fisher' } }
				link="https://www.example.com"
				readTime={ 17 }
				excerpt={ {
					rendered:
						'<p>John James "Jimbo" Fisher Jr. is an American college football coach and former player.</p>',
				} }
				embeds={ {
					author: [ { name: 'Captain Jack Sparrow' } ],
					'wp:featuredmedia': featuredMedia,
				} }
				numberFormatter={ new Intl.NumberFormat( 'en-US' ) }
			/>
		).render();

		expect( post ).toMatchSnapshot();
	} );

	it( 'Shows a post without media or read time correctly', () => {
		const post = shallow(
			<Post
				title={ { rendered: 'Jimbo Fisher' } }
				link="https://www.example.com"
				excerpt={ {
					rendered:
						'<p>John James "Jimbo" Fisher Jr. is an American college football coach and former player.</p>',
				} }
				embeds={ {
					author: [ { name: 'Captain Jack Sparrow' } ],
				} }
				numberFormatter={ new Intl.NumberFormat( 'en-US' ) }
			/>
		).render();

		expect( post ).toMatchSnapshot();
	} );

	it( 'Excludes read time meta if 0', () => {
		const post = shallow(
			<Post
				title={ { rendered: 'Jimbo Fisher' } }
				link="https://www.example.com"
				embeds={ {
					author: [ { name: 'Captain Jack Sparrow' } ],
				} }
				readTime={ 0 }
				numberFormatter={ new Intl.NumberFormat( 'en-US' ) }
			/>
		).render();

		expect( post ).toMatchSnapshot();
	} );
} );
