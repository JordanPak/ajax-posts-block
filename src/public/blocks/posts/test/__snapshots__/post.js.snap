// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Compoonent Shows post with all features correctly 1`] = `
<a
  class="apb-post"
  href="https://www.example.com"
>
  <img
    alt="Thumbnail alt text"
    class="apb-post-thumbnail"
    height="200"
    src="thumbnail.jpg"
    width="400"
  />
  <div
    class="apb-post-details"
  >
    <h3
      class="apb-post-title"
    >
      Jimbo Fisher
    </h3>
    <div
      class="apb-post-meta"
    >
      <span
        class="apb-post-author"
      >
        by Captain Jack Sparrow
      </span>
      <span
        class="apb-read-time"
      >
        17 minute read time
      </span>
    </div>
    <div
      class="apb-post-excerpt"
    >
      <p>
        John James "Jimbo" Fisher Jr. is an American college football coach and former player.
      </p>
    </div>
  </div>
</a>
`;
